src/Scientist/Internals/Experiment.cs

```diff
@@ -1,109 +1,32 @@
         {
             _name = name;
-            _control = control;
-            _candidate = candidate;
         }
 
-        public async Task<T> Run()
-        {
-            // Randomize ordering...
-            var runControlFirst = _random.Next(0, 2) == 0;
-            ExperimentResult controlResult;
-            ExperimentResult candidateResult;
-
-            if (runControlFirst)
-            {
-                controlResult = await Run(_control);
-                candidateResult = await Run(_candidate);
-            }
-            else
-            {
-                candidateResult = await Run(_candidate);
-                controlResult = await Run(_control);
-            }
-
-            // TODO: We need to compare that thrown exceptions are equivalent too https://github.com/github/scientist/blob/master/lib/scientist/observation.rb#L76
-            // TODO: We're going to have to be a bit more sophisticated about this.
-            bool success =
-                controlResult.Result == null && candidateResult.Result == null
-                || controlResult.Result != null && controlResult.Result.Equals(candidateResult.Result)
-                || controlResult.Result == null && candidateResult.Result != null;
-
-            // TODO: Get that duration!
-            var observation = new Observation(_name, success, controlResult.Duration, candidateResult.Duration);
+        public void Use(Func<Task<T>> control) { _control = control; }
 
-            // TODO: Make this Fire and forget so we don't have to wait for this
-            // to complete before we return a result
-            await Scientist.ObservationPublisher.Publish(observation);
+        public void Use(Func<T> control) { _control = () => Task.FromResult(control()); }
 
-            if (controlResult.ThrownException != null) throw controlResult.ThrownException;
-            return controlResult.Result;
-        }
+        // TODO add optional name parameter, and store all delegates into a dictionary.
 
-        static async Task<ExperimentResult> Run(Func<Task<T>> experimentCase)
-        {
-            var sw = new Stopwatch();
-            sw.Start();
-            try
-            {
-                // TODO: Refactor this into helper function?  
-                var result = await experimentCase();
-                sw.Stop();
+        public void Try(Func<Task<T>> candidate) { _candidate = candidate; }
 
-                return new ExperimentResult(result, new TimeSpan(sw.ElapsedTicks));
-            }
-            catch (Exception e)
-            {
-                sw.Stop();
-                return new ExperimentResult(e, new TimeSpan(sw.ElapsedTicks));
-            }
-        }
+        public void Try(Func<T> candidate) { _candidate = () => Task.FromResult(candidate()); }
 
-        class ExperimentResult
-        {
-            public ExperimentResult(T result, TimeSpan duration)
+        internal ExperimentInstance<T> Build()
             {
```

Comment:
This is indented oddly. In fact, you don't need the braces if you make this an expression bodied member.